### DISCUSSION TITLE
Retrieving Token and Cost Information for OpenAI's Whisper and Text-to-Speech (TTS) Models in Langsmith?

### DISCUSSION THREAD

**Discussion by talalkhan95** [langchain-ai/langsmith-sdk#822] [Q&A]
Is it possible to retrieve token and cost information for OpenAI's Audio and TTS models in LangSmith runs while tracing?
<img width="500" alt="Screenshot 2024-06-25 at 6 38 11 PM" src="https://github.com/langchain-ai/langsmith-sdk/assets/73141155/e4b1eda3-4ad5-4cee-a09f-a7a41b6187d3">

I am able to retrieve this information for chat completion models like GPT-4 and GPT-3.5 Turbo. 
<img width="500" alt="Screenshot 2024-06-25 at 6 43 22 PM" src="https://github.com/langchain-ai/langsmith-sdk/assets/73141155/a1c249ef-65c2-43c4-af0f-5fbc992c9b9b">

However, for the Audio and TTS models, only traces are being logged without any specific details.
<img width="500" alt="Screenshot 2024-06-25 at 6 43 58 PM" src="https://github.com/langchain-ai/langsmith-sdk/assets/73141155/11718921-ba71-478c-a2ba-af1918d0dbf8">

**Comment by aurelienpirot** [comment]
do you find the solution?

**Reply by zee229** [reply]
any updates?
